### ISSUE TITLE
Create admin page to accept/reject new registrations

### ISSUE THREAD

**Issue by ocrasborn** [Signbank/Global-signbank#460]
* See #355 for whole workflow

**Comment by Woseseltops** [comment]
This is the workflow @ocrasborn is talking about:

> 1. Prospective user clicks on 'registration' somewhere
> 2. User goes to page where personal details can be filled in
> 3. User checks "Accept terms of use" and clicks submit
> 4. New account is created, but waits for consent from the system manager
> 5. System manager goes to a page where a list of new request appears, with buttons 'accept' and 'reject'; reject removes the user, accept makes the user 'active'
> 6. User receives an email saying account is active, with a link to the Conditions of Use page

An attempt at organizing my thoughts about this:
* Steps 1-4 were already built by @susanodd (just tested it), so we've got that part covered.
* We can come pretty close to step 5 by simply going to the user list in the admin, and then filtering for non-active users: https://signbank.science.ru.nl/admin/auth/user/?is_active__exact=0 . The only difference is that this way we cannot make a difference between yet-to-be-activated and deactivated accounts. A solution could be to create a separate group for yet-to-be-activated accounts (something like 'aspiring', 'request' or 'candidate'), so we can filter for non-active users in this group. Activating a requested account would then mean to check the 'active' checkbox and removing the 'aspiring' group'.
* Step 6 should be possible with some kind of Django hook: custom code that is run when a user is saved in the admin ( https://stackoverflow.com/questions/11909877/django-user-save-hook ).

Proposed steps:
1. Add an 'aspiring' group that is added to self-created accounts.
2. Send an email to a user when s/he has been made active
3. If the above works, create a special view for the staff where they can get a quick overview of all 'aspiring' members, and add the 'active' status and remove the 'aspiring' group in one click.

Agree @ocrasborn ?

**Comment by susanodd** [comment]
It's possible to look at the registration profiles and look for recent activation codes.
I started trying to do this for the dataset managers when they grant access.
But then I modified the registration (as wanted by @ocrasborn) to only activate after access to a dataset has been granted. I left off the checking that it's a recent new user. But if the user is currently not active, then an email is sent and they are activated by means of granting access to the dataset.

**Comment by susanodd** [comment]
So I did step 5 via the dataset manager.

**Comment by susanodd** [comment]
Step 5 for the system admin is still needed for never activated (expired activation code) users.

**Comment by susanodd** [comment]
I suspect it might be useful (a good idea) to put a warning in the dataset manager page if they want to grant change permission to a user who doesn't yet have view permission.
That seems kind of dodgy.

**Comment by ocrasborn** [comment]
Good idea @susanodd

**Comment by ocrasborn** [comment]
@woseseltops: agree

**Comment by ocrasborn** [comment]
Hm, @Woseseltops , yes, I see, I'm not sure... Feeling of wanting to be in control? And perhaps still not seeing clearly where this will go if we have hundreds of users in due course. Let's see for a while how this works. One question remaining though: how do the access rights assigned by the dataset managers relate to the groups we have set up? Is everyone in group Guest, and gets promoted to Researcher when they get write rights for a given dataset? Otherwise wouldn't the limited Guest rights restrict the dataset object permissions?

**Comment by Woseseltops** [comment]
> And perhaps still not seeing clearly where this will go if we have hundreds of users in due course

Not fully sure, but I'd say that not having an extra high accept/reject system actually scales better than having it :). To be clear: the system we have right now is that everybody can create a 'normal' account and request access to datasets, it's just that accounts without dataset access are kind of useless. Adding an extra layer would mean adding an extra human validation step before the dataset access can even be requested, so more work for Signbank managers.

> hundreds of users 

Will there ever be hundreds of users, though? I ask because, at least for the American Signbank, I believe the idea was most users would only access the public version for which you don't need an account.

> One question remaining though: how do the access rights assigned by the dataset managers relate to the groups we have set up? Is everyone in group Guest, and gets promoted to Researcher when they get write rights for a given dataset? Otherwise wouldn't the limited Guest rights restrict the dataset object permissions?

Good point, I didn't think of this. I just did the following experiment:
1. Added myself as an owner of 'TestDatasetMicha'.
2. Created a new account testregistration2, requested 'TestDatasetMicha'.
3. As my main account (superuser), set the account testregistration2 to active.
4. Check what testregistration2 could do. Answer: nothing.
5. As my main account (manager of TestDatasetMicha), added testregistration2 to 'User who can view'.
6. Check what testregistration2 could do. Answer: nothing.
7. As my main account (superuser), changed the group of testregistration2 to Researcher.
8. Check what testregistration2 could do. Answer: view the glosses in TestDatasetMicha!

In summary: to be able to use an account, it should (1) be active, (2) be at least in the 'Researcher' group and (3) have access to at least 1 dataset.

@ocrasborn maybe we fix this a bit by making new accounts active by default, and give the guest group gloss viewing rights? Without access to datasets, these permissions are useless anyway.

**Comment by susanodd** [comment]
Hmm. It seems that there is another mission missing for user testregistraion2. (Between steps 5 and 6.)
In adminviews for GlossListView, it checks the user has permission dictionary:search_gloss.
So this should be granted for new users! Sorry about that. This seems to be separate from Dataset permissions.

**Comment by Woseseltops** [comment]
> In adminviews for GlossListView, it checks the user has permission dictionary:search_gloss.
So this should be granted for new users! 

Yeah I fixed this by adding the Researcher group.

**Comment by susanodd** [comment]
I just added that to user activation (for new users when first view dataset permission is granted).

**Comment by susanodd** [comment]
I added search_gloss permission (not Researcher group).

**Comment by susanodd** [comment]
After adding the search_gloss permission, some things work but the Search page doesn't.
I duplicated @ocrasborn's actions with a new user on my local machine,

I found the error. It was on the Search page, because of the optional Add New Sign area.
There is some javascript that adds language translations to that code, but the code isn't there for a user without change permission.

This has been fixed.

**Comment by vanlummelhuizen** [comment]
@ocrasborn close this issue?
